### DISCUSSION TITLE
Policy on Neverball

### DISCUSSION THREAD

**Discussion by limdingwen** [Homebrew/discussions#4080] [Everyday usage]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 3.6.16-97-ge76c55e
ORIGIN: https://github.com/Homebrew/brew
HEAD: e76c55e18434c870cca1de7aadf9d5218271fe2d
Last commit: 3 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: b86c42a3b72271d0ff1c7b7e6257feb63290f427
Core tap last commit: 2 days ago
Core tap branch: master
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.i1Nmj27OXY/org.xquartz:0
HOMEBREW_EDITOR: /opt/homebrew/bin/nvim
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.8 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 13.1.6 build 1316
Git: 2.32.1 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 7.79.1 => /usr/bin/curl
macOS: 12.4-arm64
CLT: 14.2.0.0.1.1668646533
Xcode: 13.4.1
Rosetta 2: false
```

### Output of `brew doctor`

```shell
﻿Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Unbrewed header files were found in /usr/local/include.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected header files:
  /usr/local/include/gnumake.h
  /usr/local/include/libtcc.h

Warning: Unbrewed static libraries were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected static libraries:
  /usr/local/lib/libtcc.a

Warning: You have uncommitted modifications to Homebrew/brew.
If this is a surprise to you, then you should stash these modifications.
Stashing returns Homebrew to a pristine state but can be undone
should you later need to do so for some reason.
  cd /opt/homebrew && git stash -u && git clean -d -f

Uncommitted files:
  ?? Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/public_suffix-4.0.6/data/list.txt

You have uncommitted modifications to Homebrew/homebrew-core.
If this is a surprise to you, then you should stash these modifications.
Stashing returns Homebrew to a pristine state but can be undone
should you later need to do so for some reason.
  cd /opt/homebrew/Library/Taps/homebrew/homebrew-core && git stash -u && git clean -d -f

Uncommitted files:
   M Formula/sshfs.rb

Warning: Some taps are not on the default git origin branch and may not receive
updates. If this is a surprise to you, check out the default branch with:
  git -C $(brew --repo ignis/local-go) checkout 

Warning: Broken symlinks were found. Remove them with `brew cleanup`:
  /opt/homebrew/lib/ruby/gems/3.1.0/bin/bundle
  /opt/homebrew/lib/ruby/gems/3.1.0/bin/bundler
```

### Description of issue

I would like to start a discussion on the policy of Neverball. It seems that currently, the app is unable to be added to Homebrew in any way.

### Option 1: Adding to Homebrew Core

In this option, we create a formula to build Neverball from source, and host the resulting binary on Homebrew infra. However, formulae are not allowed to build GUI, or app bundles, so this is not an option.

### Option 2: Adding to Homebrew Casks, from the official website

This is the correct option, however [upstream does not seem to be responding to requests](https://github.com/Neverball/neverball/issues/151) to update the official website with the latest Mac builds. The current Mac build on the official website is now so old, it no longer runs (I believe it's compiled for PowerPC).

### Option 3: Adding to Homebrew Casks, from a non-official source

This is forbidden by Cask policy due to the risk. Currently, builds by @xif-fr are hosted [here](https://github.com/Neverball/neverball/issues/151#issuecomment-502465387), and I mirror their builds [here](https://github.com/Neverball/neverball/issues/151#issuecomment-1336507994).

### Option 4: Adding Nuncabola to Homebrew Casks

This seems to be the best option from my perspective, because it contains no policy changes or exceptions. [Nuncabola](http://uppgarn.com/nuncabola/) is a Java alternative to Neverball, and is referenced by the Neverball website as an alternative to the Mac download (though why they do not host the 1.6.0 version is lost on me).

### Option 5: Adding to Homebrew Casks, using some sort of infrastructure managed by Homebrew

In this option, Homebrew maintainers will build Neverball from source and host it on a trusted infra, which will then be added to Casks. Is there any precendence of this happening?

**Comment by SMillerDev** [comment]
If upstream doesn't support the app on macOS Homebrew won't either. If it doesn't meet all the criteria for inclusion there's not much brew can do about it. I recommend contacting upstream and asking them how to install on Mac.

**Reply by limdingwen** [reply]
I'll take a look at Option 4, in that case, if I have the time.

**Comment by p-linnane** [comment]
Option 6: [Host it in your own tap](https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap).
